src/lib/metrics-gauge.ts

```diff
@@ -1,11 +1,14 @@
+import type { Logger } from './logger';
+import type { IUnleashConfig } from './types';
 import { createGauge, type Gauge } from './util/metrics';
 
 type RestrictedRecord<T extends readonly string[]> = Record<T[number], string>;
 type Query<R> = () => Promise<R | undefined | null>;
-type MapResult<R> = (result: R) => {
+type MetricValue<R> = {
     count: number;
     labels: RestrictedRecord<GaugeDefinition<R>['labelNames']>;
 };
+type MapResult<R> = (result: R) => MetricValue<R> | MetricValue<R>[];
```

Comment:
A single task can output different counts for different labels. This is useful for cases like `client_apps_total` at line 40